jsurfer-core/src/main/java/org/jsfr/json/compiler/JsonPathCompiler.java

```diff
@@ -447,6 +483,27 @@ private static void arrayWildcard(String key, JsonPathFilter filter, JsonPath.Bu
         builder.arrayWildcard(key, filter);
     }
 
+    private static ItemMethod getItemMethod(JsonPathParser.ItemMethodContext ctx) {
+        String methodName = ctx.KEY().getText();
+        ItemMethod itemMethod = ItemMethod.from(methodName);
+        if (itemMethod == null) {
+            throw new InputMismatchException(String.format("Invalid item method %s. Supported: %s", methodName,
+                Arrays.toString(ItemMethod.values())));
+        }
+        return itemMethod;
+    }
+
+    private static Type getType(TerminalNode quoteString) {
+        String typeName = unescapeString(quoteString.getText());
+        Type type = Type.from(typeName);
+        if (type == null) {
+            throw new InputMismatchException(
```

Comment:
We shouldn't throw in this case. If the user does `foo.type() == 'non-existing-type'`, it can simply return no results, perhaps with some always-false filter.

We should also cover it in tests.

Reply:
Done